src/vs/editor/common/config/editorOptions.ts

```diff
@@ -2946,6 +2952,11 @@ class EditorInlayHints extends BaseEditorOption<EditorOption.inlayHints, IEditor
 					type: 'boolean',
 					default: defaults.padding,
 					description: nls.localize('inlayHints.padding', "Enables the padding around the inlay hints in the editor.")
+				},
+				'editor.inlayHints.maximumLength': {
+					type: 'number',
+					default: defaults.maximumLength,
+					description: nls.localize('inlayHints.maximumLength', "Maximum length for inlay hints. Set to 0 to have an unlimited length.")
```

Comment:
Maybe update the description to mention what this is for, like "Maximum length for inlay hints before they get truncated by the editor"

It should maybe also mention that this limit allows to all inlay hint for a line and not to individual inlay hints. That tbh is the weird part of my implementation and maybe the setting needs to be 2 dimensional, e.g the max per inlay and the overall max

Reply:
I think I got it now, tested even edge case when it's 1 character away so it doesn't generate 2 ellipsis and it seems fine:

![image](https://github.com/microsoft/vscode/assets/11574755/5879099b-ffff-4d26-95de-48147e148c5c)
![image](https://github.com/microsoft/vscode/assets/11574755/8fa0d02d-b22e-4056-b391-5eb536b037c4)
![image](https://github.com/microsoft/vscode/assets/11574755/cc12ae63-12a1-4b3b-87c8-83ec0e85e547)
![image](https://github.com/microsoft/vscode/assets/11574755/1c4d247a-9b23-450d-b88d-eca213c62f4d)
![image](https://github.com/microsoft/vscode/assets/11574755/e4b866f7-7874-43aa-b9c3-b3125c17aa2d)